API/src/main/java/xyz/kyngs/librelogin/api/server/ServerHandler.java

```diff
@@ -29,6 +30,17 @@ public interface ServerHandler<P, S> {
      */
     ServerPing getLatestPing(S server);
 
+    /**
+     * Chooses an optimal lobby server to connect the player to. Usually the one with the lowest player count.
+     *
+     * @param user     The user of the player, or null if the player is from bedrock
+     * @param player   The player we're choosing the server for
+     * @param remember Whether to respect the remember last server option
+     * @param fallback Whether to select fallback server or not
+     * @return An optimal lobby server, or null if there are no lobby servers
+     */
+    S chooseLobbyServer(@Nullable User user, P player, boolean remember, boolean fallback);
```

Comment:
The return suggests that if there are no lobbies, null is returned. 
However, the implementation does the opposite:
If there are no lobbies, NoSuchElementException is thrown. If the event was canceled, null is returned. 

Perhaps some unchecked exception like CancelledEventException shall be thrown and documented to indicate that the event was canceled? In theory, this is a breaking change but shouldn't occur unless some other plugin cancels the event.

Reply:
Fixed this in [9e7d402](https://github.com/kyngs/LibreLogin/pull/103/commits/9e7d4023429dcf904d03cd0887faf67d7ffdbfb2)